app/controllers/application_controller.rb

```diff
@@ -13,8 +13,21 @@ def logged_in?
 
   def require_user
     unless logged_in?
-      flash[:info] = "Access reserved for members only. Please sign in first."
+      flash[:error] = "Access reserved for members only. Please sign in first."
       redirect_to root_path
     end
   end
+
+  private
+
+  def set_reviews(object)
```

Comment:
Setting the average and calculating is the work for model. Watch solution videos on where to place this functionality. Also for average rating you can use `ActiveRecords#average`.